scribus/plugins/import/pdf/slaoutput.cpp

```diff
@@ -3896,3 +3926,680 @@ bool SlaOutputDev::checkClip()
+				if ((textRegionLines.size() >= 2)) //TODO: Need to setup some parameters replating to width matching, they mainly relate to justfication  && closeToX(textRegionLines[textRegionLines.size() - 2].baseOrigin.x() +  textRegionLines[textRegionLines.size() - 2].width, maxWidth))
+				{
+					//TODO: add a new line and update the deltas
+					pass = FrameworkLineTests::NEWLINE;
+#ifdef DEBUG_TEXT_IMPORT
+					qDebug() << "NEWLINE1 point:" << point << " _lastXY:" << lastXY << " origin: " << textRegioBasenOrigin <<  " this:" << this << " linespacing: " << lineSpacing;
+
+#endif // DEBUG
+				}   // we only have the first line so far, so pass without much of a test.
+				else if (textRegionLines.size() == 1) 
+				{
+					pass = FrameworkLineTests::NEWLINE;
+#ifdef DEBUG_TEXT_IMPORT
+					qDebug() << "NEWLINE2 point:" << point << " _lastXY:" << lastXY << " origin: " << textRegioBasenOrigin << " this:" << this << " linespacing: " << lineSpacing;
+#endif
+				}
+			}
+		}
+		else
+		{
+#ifdef DEBUG_TEXT_IMPORT
+			qDebug() << "NEWLINE2 oops:"<<point << ":"<<textRegioBasenOrigin << ":" << lineSpacing;
+#endif
+		}
+	}
+	return pass;
+}
+
+// Just perform some basic checks to see if newPoint can reasonably be asscribed to the current textframe.
+TextRegion::FrameworkLineTests TextRegion::isRegionConcurrent(QPointF newPoint)
+{	
+	if (glyphs.empty())
+	{
+		lineBaseXY = newPoint;
+		lastXY = newPoint;
+	}
+
+	//TODO: I need to write down which ones we want so I can work it all out, for now just some basic fuzzy matching support.
+	bool xInLimits = false;
+	if (closeToX(newPoint.x(), lastXY.x()))
+	{
+		xInLimits = true;
+	}
+	else
+	{
+		qDebug() << "X out of limits " << lastXY << " : " << newPoint;
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		yInLimits = true;
+	}
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);
+	return pass;
+}
+
+
+TextRegion::FrameworkLineTests TextRegion::moveToPoint(QPointF newPoint)
+{
+	//
+	//qDebug() << "moveToPoint: " << newPoint;
+	// Do some initilization if we are in a new text region
+	// we could also update these if glyphindex = -1;
+	if (glyphs.empty())
+	{
+		lineBaseXY = newPoint;
+		lastXY = newPoint;
+		//qDebug() << "newPoint3: " << newPoint;
+	}
+	
+	//TODO: I need to write down which ones we want so I can work it all out, for now just some basic fuzzy matching support.
+	//TODO: x limiting should be different for moving as opposed to adding a new glyph because moiving is due to a discontinuity in glyphs
+	bool xInLimits = false;
+	if (closeToX(newPoint.x(), lastXY.x()))
+	{
+		//qDebug() << "newPoint6: " << newPoint;
+		xInLimits = true;
+	}
+	bool yInLimits = false;
+	if (closeToY(newPoint.y(), lastXY.y()))
+	{
+		//qDebug() << "newPoint7: " << newPoint;
+		yInLimits = true;
+	}
+	//qDebug() << "newPoint8: " << newPoint;
+	FrameworkLineTests pass = linearTest(newPoint, xInLimits, yInLimits);

```

Comment:
same as above... you need one to three lines here...instead of a dozen

Reply:
and if it does the same as above, you should think of using the function above : - )